uberpoet/loccalc.py

```diff
@@ -0,0 +1,49 @@
+#  Copyright (c) 2021 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import tempfile
+from math import ceil
+from os.path import join
+
+from .commandlineutil import count_loc
+from .filegen import Language
+from .memoize import memoized
+
+
+class LOCCalculator(object):
+
+    @memoized
+    def calculate_loc(self, file_gen):
+        # actual code = lines of code, minus whitespace
+        # calculated using cloc
+        file_result = file_gen.gen_file(3, 3)
```

Comment:
I don't think putting in a file generator class and generating a file independently with one static size is a good way to go about this.   This should be able to take any string that represents code and tell you the loc count for it.  What if you wanted to make bigger files in the future, or make that an option?  I don't think someone passing in a file generator object is expecting the LOC count to just be one size.

Reply:
that is true, this used to be doing this before but i see your point on making such assumption now with this reorganization

Reply:
will change to take a string.

Reply:
updated